vortex-array/src/arrays/primitive/compute/is_constant.rs

```diff
@@ -9,24 +11,49 @@ impl IsConstantFn<&PrimitiveArray> for PrimitiveEncoding {
         &self,
         array: &PrimitiveArray,
         _opts: &IsConstantOpts,
-    ) -> vortex_error::VortexResult<Option<bool>> {
-        let is_constant = match_each_native_ptype!(array.ptype(), |$P| {
-            compute_is_constant(array.as_slice::<$P>())
+    ) -> VortexResult<Option<bool>> {
+        let is_constant = match_each_native_ptype!(array.ptype(), integral: |$P| {
+            compute_is_constant::<_, {256 / size_of::<$P>()}>(array.as_slice::<$P>())
```

Comment:
1. 256 is bits and size_of is bytes, which seems like the wrong size?
2. I think that instead of hardcoding 256 you can probably make an educated guess using `if_cfg!`, I remember Arrow having a good example of that somewhere (`aggregate`?)

Reply:
This was meant as a byte size. I need to spelunk in llvm for a threshold that I observed when benchmarking

Reply:
which one?

Reply:
there's 3 kind of assembly I saw generated for equals logic - 1. scalar value by value, 2. comparison using simd registers, 3. comparison using libc memcmp. I'm trying to figure out if there's a difference between 2 and 3. Looks like libc memcmp does a bit of extra work for fast comparison

Reply:
Ok https://github.com/rust-lang/rust/issues/41638 is what I found. I think we want to end up using the small array version of the code that doesn't call memcmp

Reply:
though if the array is small we could just do a one big memcmp